263 Feature Request: Scenario Template & Contribution Guide

TeneBrae93 opened the pull request:
#### Overview of Changes
- Created a scenario template with basic resources, output, and whitelisting to provide a starting place for CloudGoat contributors. I also provided comments explaining some of the main things to keep in mind. 
- Updated the Contribution Guidelines on the main README to help people get started with contributing and linked them to the scenario template as a starting point

andrew-aiken commented on scenarios/scenario_template/cheat_sheet.md:
```diff
@@ -0,0 +1,10 @@
+# [SERVICE]_Secrets Walkthrough
```
Was `_Secrets` meant to be left in? Not sure if this was carried over from sns_secrets

TeneBrae93 replied:
Oh good catch, I missed that one. Just fixed it

andrew-aiken commented on README.md:
```diff
@@ -328,13 +328,30 @@ For features, much the same applies! Be specific in your request, and make sure
-2. Follow good git practices: use pull requests, prefer feature branches, always write clear commit messages.
-3. CloudGoat uses `black` and `flake8` - Python syntax and style linters - If you're going to commit code for CloudGoat, ensure that first `flake8`, and then `black` are both run on all Python files in `core/python/` and on `cloudgoat.py`. `black`'s decisions take priority over `flake8`'s. Both of these are commented out in the `core/python/requirements.txt` file since normal users don't need the ...
-4. CloudGoat code should always use the BSD 3-clause license.
+1. **Creating a New Scenario**:
```
😍

TeneBrae93 replied:
Hahaha glad you like it!

andrew-aiken reviewed (changes requested):
Overall looks good

There already is a "hidden" template directory
https://github.com/RhinoSecurityLabs/cloudgoat/tree/master/core/terraform/example-scenario

TeneBrae93 commented:
Good catch @andrew-aiken on the existing template (didn't even realize that) - and neither did the people I talked to at Rhino when we discussed this. It's also pretty light on details (doesn't show whitelisting a public resource like an EC2 which is a specific problem we ran into with a community-submitted scenario - the EC2 is public on the internet). 

I added a new commit that just removes this old one entirely, and puts this new one in the scenario folder. I think this makes it easier for people to find; also by adding it to the main README's contribution guidelines, this one should be less hidden. 

Finally, I noticed some things I missed (manifest.yml in particular), so I added that to the new scenario template.

andrew-aiken reviewed (approved):
Awesome
Thanks for improving the readme, should make it a lot easier for people to create scenarios

nobodynate reviewed (approved):
Looks good!
